pytorch_lightning/loops/optimization/closure.py

```diff
@@ -122,69 +56,10 @@ def consume_result(self) -> ClosureResult:
+    def closure(self, *args: Any, **kwargs: Any) -> T:
         """Implements the behavior of the closure once it is getting called."""
         pass
 
-    def __call__(self, *args: Any, **kwargs: Any) -> Optional[Tensor]:
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
         self._result = self.closure(*args, **kwargs)
-        return self._result.loss
-
-
-class Closure(AbstractClosure):
-    """An implementation of a :class:`AbstractClosure` for optimization in Lightning that combines three elementary
-    closures into one: ``training_step``, ``backward`` and ``zero_grad``.
-
-    The Closure gets created by the training loop(s) and is then passed to the
-    :meth:`torch.optim.Optimizer.step` method. An optimizer is responsible for calling the closure and optionally
-    do something with the output.
-
-    Args:
-        step_fn: This is typically the :meth:`pytorch_lightning.core.lightning.LightningModule.training_step
-            wrapped with processing for its outputs
-        backward_fn: A function that takes a loss value as input, performs back-propagation and returns the loss value.
-            Can be set to ``None`` to skip the backward operation.
-        zero_grad_fn: A function that zeroes the gradients. Can be set to ``None`` to skip zero_grad, for example
-            when accumulating gradients.
-        profiler: A profiler for profiling the actions of the passed in closure functions.
-
-    Example:
-
-        closure = Closure()
-        optimizer = torch.optim.Adam(...)
-        optimizer.step(closure)
-    """
-
-    warning_cache = WarningCache()
-
-    def __init__(
-        self,
-        step_fn: Callable[[], ClosureResult],
-        backward_fn: Optional[Callable[[Tensor], Tensor]] = None,
-        zero_grad_fn: Optional[Callable[[], None]] = None,
-        profiler: Optional[BaseProfiler] = None,
-    ):
-        super().__init__()
-        self._step_fn = step_fn
-        self._backward_fn = backward_fn
-        self._zero_grad_fn = zero_grad_fn
-        self._profiler = PassThroughProfiler() if profiler is None else profiler
-
-    def closure(self, *args: Any, **kwargs: Any) -> ClosureResult:
-        with self._profiler.profile("training_step_and_backward"):
-            step_output = self._step_fn()
-
-            if step_output.closure_loss is None:
-                self.warning_cache.warn(
-                    "`training_step` returned `None`. If this was on purpose, ignore this warning..."
-                )
-
-            if self._zero_grad_fn is not None:
-                with self._profiler.profile("zero_grad"):
-                    self._zero_grad_fn()
-
-            if self._backward_fn is not None and step_output.closure_loss is not None:
-                with self._profiler.profile("backward"):
-                    step_output.closure_loss = self._backward_fn(step_output.closure_loss)
-
-        return step_output
+        return self
```

Comment:
return self? hmm?

Reply:
I chose to return self to chain

`closure().consume_result()`

But it's unused. It can also return None